lib/output.mli

```diff
@@ -16,15 +16,19 @@
 
 (** Test outputs. *)
 
-type t = [ `Output of string | `Ellipsis ]
+type sub = S of string | Sub_ellipsis
+
+type t = [ `Output of sub list | `Ellipsis ]
 (** The type for test outputs. Ellipisis ([...]) allow to skip zero,
-   one or many lines while comparing the outputs with {!equal}. *)
+    one or many lines while comparing the outputs with {!equal}.
+    Sub_ellipsis ([<...>]) allow to skip zero, one or more characters or words
+    in e line, while comparing the outputs ith {!equal}. *)
 
-val equal : t list -> t list -> bool
-(** [equal x y] is true iff [x] and [y] are equivalent, modulo
+val equal : t list -> ref:t list -> bool
```

Comment:
I'm a bit surprised this compares two `t list` rather than a `string list` to a `t list`, do you happen to know why?

Reply:
it's because the arguments are swapped when calling the function recursively, I don't know how to implement it differently, so I added new functions operating on `string list` and calling the initial functions